jobs/async-upload/job/models.py

```diff
@@ -96,18 +96,30 @@ def validate_uri_storage(self) -> 'URISourceStorageConfig':
 SourceConfig = Union[S3StorageConfig, OCIStorageConfig, URISourceStorageConfig]
 DestinationConfig = Union[S3StorageConfig, OCIStorageConfig]
 
+class UploadIntent(StrEnum):
+    create_model = "create_model"
+    create_version = "create_version"
+    update_artifact = "update_artifact"
 
 class ModelConfig(BaseModel):
     """Model registry model information."""
-    id: str = Field(..., description="Model ID")
-    version_id: str = Field(..., description="Model version ID")
-    artifact_id: str = Field(..., description="Model artifact ID")
+    upload_intent: UploadIntent = Field(UploadIntent.update_artifact, description="Model upload intent")
+    id: str | None = Field(..., description="Model ID")
+    version_id: str | None = Field(..., description="Model version ID")
+    artifact_id: str | None = Field(..., description="Model artifact ID")
 
     @model_validator(mode='after')
     def validate_model_ids(self) -> 'ModelConfig':
         """Validate that all model IDs are provided."""
-        if not all([self.id, self.version_id, self.artifact_id]):
-            raise ValueError("Model ID, version ID and artifact ID must be set")
+        if self.upload_intent == UploadIntent.create_model:
+            if not all([self.id, self.version_id, self.artifact_id]):
+                raise ValueError("Model ID, Model Version ID and Model Artifact ID must be set when intent is create_model")
+        elif self.upload_intent == UploadIntent.create_version:
+            if not all([self.id, self.version_id]):
+                raise ValueError("Model ID and version ID must be set when intent is create_version")
```

Comment:
I'm not sure I'm following this one, if it's meant to "Create Version",
how can the version_id be known before-hand? 🤔